docs/midstream-getting-serveless-operator-bugfixes.md

```diff
@@ -0,0 +1,132 @@
+When we need to fix a bug, we overwrite the existing image with a new one.
+
+For example, when we identify a bug in 1.24 version, we change the code in 1.24 branch and the CI/CD overwrites the `1.24.0` image.
+
+## 2. Prerequisites
+
+- Usage of the midstream Serverless Operator and not the fully released product
+- Access to OCP cluster with installed Serverless Operator
+- Permission to delete pods and replicasets in `openshift-serverless` namespace.
+- If using Knative Eventing and Knative Kafka components, permission to delete pods and replicasets in `knative-eventing` namespace.
+- If using Knative Serving, permission to delete pods and replicasets in `knative-serving`, `knative-serving-ingress` namespaces.
+
+## 3. Execute/Resolution
+
+1. Get the Serverless Operator images used in the OpenShift cluster right now, for later comparison:
+  ```shell
+  > kubectl get pods -n openshift-serverless -o jsonpath='{range .items[*]}{"\n"}{.metadata.name}{": "}{range .status.containerStatuses[*]}{.imageID}{end}{end}' | sort 
+  knative-openshift-5dbb4bd4b7-q6wdp: registry.ci.openshift.org/knative/openshift-serverless-v1.24.0@sha256:26c3127e3dab1e102b94c305916d405b57d397c56348e0d73c7e1c2cddf3310b
+  knative-openshift-ingress-78766bdc5c-zfzfl: registry.ci.openshift.org/knative/openshift-serverless-v1.24.0@sha256:7a417307328f76462560c5d4f814566135e59fc1b20758be720d090047ec682e
+  knative-operator-65487bf7fc-vbgkk: registry.ci.openshift.org/knative/openshift-serverless-v1.24.0@sha256:589735abf033c61247e594f3fba9e7d87ac1093d1ebc0abe73f825944fe5e465
+  ```
+2. Delete Serverless Operator pods to force them to be recreated with the new image (we explicitly set `imagePullPolicy` to `Always`):
+  ```shell
+  > kubectl delete pods -n openshift-serverless -l 'name in (knative-operator, knative-openshift, knative-openshift-ingress)'
+  pod "knative-openshift-555d4d98d7-8ns5b" deleted
+  pod "knative-openshift-ingress-7bc7c7b47b-gc5c9" deleted
+  pod "knative-operator-67c4958cc6-zg8b9" deleted
+  ```
+
+3. If using Knative Serving, delete Knative Serving control plane replicasets. They will be recreated by the operator:
+  ```shell
+  > kubectl delete replicasets.apps -n knative-serving -l 'app in (controller, webhook)'
+  replicaset.apps "controller-57466669cf" deleted
+  replicaset.apps "controller-7bb7748cd8" deleted
+  replicaset.apps "webhook-6cb8b848bd" deleted
+  ```
+
+4. If using Knative Serving, delete jobs as their images cannot be mutated:
+  ```shell
+  > kubectl delete jobs -n knative-serving -l 'app in (storage-version-migration-serving)'
+  job.batch "storage-version-migration-serving-serving-1.3.0" deleted
+  ```
+
+5. If using Knative Serving, delete Knative Serving Ingress control plane replicasets. They will be recreated by the operator:
+  ```shell
+  > kubectl delete replicasets.apps -n knative-serving-ingress -l 'app in (net-kourier-controller)'
+  replicaset.apps "net-kourier-controller-84d4b75589" deleted
+  ```
```

Comment:
These could all be indented into one sub-section so if the reader is not using Knative Serving they can skip over the entire section.

Simarly for the following sections around Eventing and Kafka components.

Reply:
would it make sense to break these into different files, like chapters?

Reply:
We now have separate subsections. Not sure if it is a good idea to use different pages/files.

In the end, IMO it is better to have a single page for the person who is doing the update.